vagrant/Vagrantfile

```diff
@@ -0,0 +1,26 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+Vagrant.configure("2") do |config|
+
+    # Ubuntu 16.04 LTS
+    config.vm.box = "fosskers/docker-1.12"
+
+    # Docker permissions are available for `ubuntu` user only.
+    config.ssh.username = "ubuntu"
+
+    # Ports to the services
+    config.vm.network :forwarded_port, guest: 8080, host: 8080
+    config.vm.network :forwarded_port, guest: 9999, host: 9999
+    config.vm.network :forwarded_port, guest: 8777, host: 8777
+
+    # VM resource settings
+    config.vm.provider :virtualbox do |vb|
+        vb.memory = 12288
+        vb.cpus = 4
+    end
+
+    # Provisioning
+    config.vm.provision :shell, path: "bootstrap.sh"
```

Comment:
I think the primary reason the Bash provisioning is lean is because a significant portion of it was done during the Vagrant box creation process. It is also not clear why all of the other dependencies are containerized, but the JDK still remains a dependency on the host.

I also disagree with the idea that Vagrant lacks value when interacting with the Docker Engine and Docker Compose. Sure, using Docker Compose from a Linux host is faster, but that tradeoffs comes with a bunch of cons as well.

- What version are you running? Is that version the same as everyone else on the team?
- Docker for Mac actually runs within a VM; what differences crop up when a developer workstation is Linux based vs. macOS based?
- Will it be the same the next time we touch this project in 6 months?
- How to we synchronize version bumps of the engine with ECS?

Again, all of these questions have proven answers when Vagrant is used to encapsulate development dependencies. Also, we got to this solution by enduring all of the cons associated with running Docker directly on people's workstations.